strprofiler/shiny_app/clastr_api.py

```diff
@@ -4,7 +4,7 @@
 from flatten_json import flatten
 
 
-def clastr_query(query, query_filter, include_amelogenin, score_filter):
+def _clastr_query(query, query_filter, include_amelogenin, score_filter):
```

Comment:
I think we actually should make these public and add doc strings to them. Might be useful generally in other applications as well. Should probably move this module outside of the shiny app package and into the main `strprofiler` one as well.

Reply:
That makes sense, I will make those changes and draft doc strings. 

Also, the py-shiny team just pushed a feature https://github.com/posit-dev/py-shiny/pull/918 that I would like to explore adding in certain places. The API call in particular seems a logical place to add this feature in.

Reply:
Oh, that's nice. Yeah, I will probably try to add the CLASTR API functionality to the CLI interface at some point (probably not for next release).

There are probably one or two other functions that I made malformed or lazy docstrings for that I'll fix before release as well. Format should be similar to the `strprofiler` one for sphinx to render them properly for ReadTheDocs.

Reply:
I was thinking about this a bit more. The functions as currently written return a specific table (i.e., `_clastr_query`) for display and raw xlsx bytes (i.e., `_clastr_batch_query`) to hold in memory until download is requested. I view these functions as similar to the adjustments done for the single and batch queries in `strprofiler/shiny_app/calc_functions.py`. For global CLASTR functions, I am planning to write new functions in the main `strprofiler` that return XLSX file output from the POST results. Doing that would simplify the functions currently written (no JSON parsing needed), and allow for generalized use. Does that seem acceptable?

Reply:
That sounds good to me.